app/src/main/java/com/zeapo/pwdstore/repository/GitRepositoryImpl.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+package com.zeapo.pwdstore.repository
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.preference.PreferenceManager
+import com.zeapo.pwdstore.utils.PasswordItem
+import java.io.File
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+
+class GitRepository : PasswordRepository {
+    private var repository: Repository? = null
+    private var settings: SharedPreferences? = null
+
+    override fun initialize(context: Context) {
+        settings = PreferenceManager.getDefaultSharedPreferences(context.applicationContext)
+        val dir = File(getRepositoryDirectory(context), ".git")
+        if (!dir.exists() && !dir.mkdir()) throw RuntimeException("Failed to create repository directory")
+        val builder = FileRepositoryBuilder()
+        repository = try {
+            builder.setGitDir(dir)
+                    .readEnvironment()
+                    .build()
+        } catch (e: Exception) {
+            e.printStackTrace()
+            null
+        }
+    }
+
+    override fun createRepository(context: Context, dir: File) {
+        dir.deleteRecursively()
+        Git.init().setDirectory(dir).call()
+        initialize(context)
+    }
+
+    override fun getRepository(): Repository? = repository
+
+    override fun getFilesList(path: File): ArrayList<File> {
+        if (!path.exists()) return arrayListOf()
+        return path.listFiles { file -> file.isDirectory || file.extension == ".gpg" }
+                ?.toCollection(ArrayList()) ?: arrayListOf()
+    }
+
+    override fun listPasswords(path: File, rootDir: File, sortOrder: PasswordSortOrder): ArrayList<PasswordItem> {
```

Comment:
Would it be worth it to extract the similar logic in SearchableRepositoryViewModel and reuse it here?

Reply:
Probably yeah. I'll look into it once the no-git implementation is done.